url-gen/src/main/kotlin/com/cloudinary/asset/Asset.kt

```diff
@@ -162,7 +162,11 @@ abstract class BaseAsset constructor(
                 mutableSource
             ).joinToString("/").cldMergeSlashedInUrl()
 
-        val analytics = if (urlConfig.analytics) "_a=${generateAnalyticsSignature()}" else null
+        var urlObject = URL(url)
+        val analytics = if (urlConfig.analytics && cloudConfig.authToken == null && urlObject.query == null)
+            "_a=${generateAnalyticsSignature()}"
+        else null
```

Comment:
indentation is a bit unclear..., if the condition becomes too long maybe switch to classic `if else`, or extract the condition to a variable

Reply:
Done